### PR TITLE
Fix replay tests

### LINKED ISSUE #78: Numerous errors logged when running “python test_replays/test_all.py”
I just forked from https://github.com/ggtracker/sc2reader (the upstream branch) on Windows 10 (Python 3.6.8) and ran test_all.py
The error message is as follows：

```
morestuff
morestuff
....Unable to change type of None [0] to 686 [frame 2135]; unit type not found in build 38215
test_replays/3.0.0.38215/fourth.SC2Replay
Release String: 3.0.0.38215
        Player 1 - Owl (Terran)
        Player 2 - dremax (Zerg)
12948   Owl     Missing ability 3B20 from Build
Owl 169.6294859359851
.Fetching remote resource: http://EU.depot.battle.net:1119/5984441b6318a28781f01f3efb54c7614af7eb286ae1c068d0f06e471fb40370.s2ma
.DOING 1
Unable to change type of None [240001] to ShakurasVespeneGeyser [frame 0]; unit type not found in build base
Unable to change type of None [280001] to ShakurasVespeneGeyser [frame 0]; unit type not found in build base
Unable to change type of None [480001] to ShakurasVespeneGeyser [frame 0]; unit type not found in build base
Unable to change type of None [4C0001] to ShakurasVespeneGeyser [frame 0]; unit type not found in build base
Unable to change type of None [500001] to CollapsiblePurifierTowerDiagonal [frame 0]; unit type not found in build base
Unable to change type of None [780001] to ShakurasVespeneGeyser [frame 0]; unit type not found in build base
Unable to change type of None [7C0001] to ShakurasVespeneGeyser [frame 0]; unit type not found in build base
Unable to change type of None [B40001] to ShakurasVespeneGeyser [frame 0]; unit type not found in build base
Unable to change type of None [B80001] to ShakurasVespeneGeyser [frame 0]; unit type not found in build base
Unable to change type of None [DC0001] to ShakurasVespeneGeyser [frame 0]; unit type not found in build base
Unable to change type of None [E00001] to ShakurasVespeneGeyser [frame 0]; unit type not found in build base
Unable to change type of None [11C0001] to ShakurasVespeneGeyser [frame 0]; unit type not found in build base
Unable to change type of None [1200001] to ShakurasVespeneGeyser [frame 0]; unit type not found in build base
Unable to change type of None [1440001] to ShakurasVespeneGeyser [frame 0]; unit type not found in build base
Unable to change type of None [1480001] to ShakurasVespeneGeyser [frame 0]; unit type not found in build base
Unable to change type of None [1580001] to ShakurasVespeneGeyser [frame 0]; unit type not found in build base
Unable to change type of None [15C0001] to ShakurasVespeneGeyser [frame 0]; unit type not found in build base
Unable to change type of None [2440001] to ShakurasVespeneGeyser [frame 0]; unit type not found in build base
Unable to change type of None [2480001] to ShakurasVespeneGeyser [frame 0]; unit type not found in build base
Unable to change type of None [24C0001] to ShakurasVespeneGeyser [frame 0]; unit type not found in build base
Unable to change type of None [2500001] to ShakurasVespeneGeyser [frame 0]; unit type not found in build base
Unable to change type of None [2540001] to ShakurasVespeneGeyser [frame 0]; unit type not found in build base
Unable to change type of None [2580001] to ShakurasVespeneGeyser [frame 0]; unit type not found in build base
Unable to change type of None [25C0001] to ShakurasVespeneGeyser [frame 0]; unit type not found in build base
Unable to change type of None [2600001] to ShakurasVespeneGeyser [frame 0]; unit type not found in build base
Unable to change type of None [2640001] to ShakurasVespeneGeyser [frame 0]; unit type not found in build base
Unable to change type of None [2680001] to ShakurasVespeneGeyser [frame 0]; unit type not found in build base
Unable to change type of None [26C0001] to ShakurasVespeneGeyser [frame 0]; unit type not found in build base
Unable to change type of None [2700001] to ShakurasVespeneGeyser [frame 0]; unit type not found in build base
Unable to change type of None [2740001] to ShakurasVespeneGeyser [frame 0]; unit type not found in build base
Unable to change type of None [2780001] to ShakurasVespeneGeyser [frame 0]; unit type not found in build base
Unable to change type of None [27C0001] to ShakurasVespeneGeyser [frame 0]; unit type not found in build base
Unable to change type of None [2800001] to ShakurasVespeneGeyser [frame 0]; unit type not found in build base
Unable to change type of None [2840001] to CollapsiblePurifierTowerDiagonal [frame 0]; unit type not found in build base
DOING 2
test_replays/3.1.0/2.SC2Replay
Release String: 3.1.0.39576
        Player 1 - OneScotch (Terran)
        Player 2 - TubbyJeff (Zerg)
18538   OneScotch       Missing ability 47E0 from Build
18976   OneScotch       Missing ability 4800 from Build
DOING 3
DOING 4
.Unknown attribute id: 5005
Unknown attribute id: 5005
Unknown attribute id: 3016
Unknown attribute id: 3016
Unknown attribute id: 5002
Unknown attribute id: 5002
Unknown attribute id: 5004
Unknown attribute id: 5004
Unknown attribute id: 5003
Unknown attribute id: 5003
Unknown attribute id: 5001
Unknown attribute id: 5001
Unknown attribute id: 3017
Unknown attribute id: 3017
Unknown attribute id: 5000
Unknown attribute id: 5000
.Unknown attribute id: 5005
.......
Unable to change type of None [2440001] to ShakurasVespeneGeyser [frame 0]; unit type not found in build base
Unable to change type of None [2480001] to ShakurasVespeneGeyser [frame 0]; unit type not found in build base
Unable to change type of None [24C0001] to ShakurasVespeneGeyser [frame 0]; unit type not found in build base
Unable to change type of None [2500001] to ShakurasVespeneGeyser [frame 0]; unit type not found in build base
Unable to change type of None [2540001] to ShakurasVespeneGeyser [frame 0]; unit type not found in build base
Unable to change type of None [2580001] to ShakurasVespeneGeyser [frame 0]; unit type not found in build base
Unable to change type of None [25C0001] to ShakurasVespeneGeyser [frame 0]; unit type not found in build base
Unable to change type of None [2600001] to ShakurasVespeneGeyser [frame 0]; unit type not found in build base
Unable to change type of None [2640001] to ShakurasVespeneGeyser [frame 0]; unit type not found in build base
Unable to change type of None [2680001] to ShakurasVespeneGeyser [frame 0]; unit type not found in build base
Unable to change type of None [26C0001] to ShakurasVespeneGeyser [frame 0]; unit type not found in build base
Unable to change type of None [2700001] to ShakurasVespeneGeyser [frame 0]; unit type not found in build base
Unable to change type of None [2740001] to ShakurasVespeneGeyser [frame 0]; unit type not found in build base
Unable to change type of None [2780001] to ShakurasVespeneGeyser [frame 0]; unit type not found in build base
Unable to change type of None [27C0001] to ShakurasVespeneGeyser [frame 0]; unit type not found in build base
Unable to change type of None [2800001] to ShakurasVespeneGeyser [frame 0]; unit type not found in build base
Unable to change type of None [2840001] to CollapsiblePurifierTowerDiagonal [frame 0]; unit type not found in build base
Unable to change type of Pylon [5840001] to PylonOvercharged [frame 10843]; unit type not found in build base
.......
test_replays/3.4.0/issueYY.SC2Replay
Release String: 3.4.0.44401
        Player 1 - bacon (Protoss)
        Player 2 - FlyingCake (Terran)
14      bacon   Missing ability 1540 from Build
23      FlyingCake      Missing ability 12C0 from Build
184     FlyingCake      Missing ability 12C0 from Build
392     FlyingCake      Missing ability F81 from Build
417     bacon   Missing ability 14A1 from Build
489     bacon   Missing ability 1540 from Build
508     FlyingCake      Missing ability 12C0 from Build
860     bacon   Missing ability 14A3 from Build
888     FlyingCake      Missing ability F83 from Build
914     bacon   Missing ability 1540 from Build
932     FlyingCake      Missing ability F83 from Build
951     FlyingCake      Missing ability 12C0 from Build
1102    bacon   Missing ability 14A2 from Build
1132    FlyingCake      Missing ability 12C0 from Build
1227    bacon   Missing ability 1540 from Build
........
Bad pid (4) for event <class 'sc2reader.events.tracker.PlayerStatsEvent'> at 11.20 [10880].
Bad pid (4) for event <class 'sc2reader.events.tracker.PlayerStatsEvent'> at 11.30 [11040].
Bad pid (4) for event <class 'sc2reader.events.tracker.PlayerStatsEvent'> at 11.40 [11200].
Bad pid (4) for event <class 'sc2reader.events.tracker.PlayerStatsEvent'> at 11.50 [11360].
Bad pid (4) for event <class 'sc2reader.events.tracker.PlayerStatsEvent'> at 12.00 [11520].
Bad pid (4) for event <class 'sc2reader.events.tracker.PlayerStatsEvent'> at 12.10 [11680].
Bad pid (4) for event <class 'sc2reader.events.tracker.PlayerStatsEvent'> at 12.20 [11840].
Bad pid (4) for event <class 'sc2reader.events.tracker.PlayerStatsEvent'> at 12.30 [12000].
Bad pid (4) for event <class 'sc2reader.events.tracker.PlayerStatsEvent'> at 12.37 [12114].
x.............
======================================================================
ERROR: test_lotv_creepTracker (__main__.TestReplays)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test_replays/test_all.py", line 437, in test_lotv_creepTracker
    replay =factory.load_replay(replayfilename,engine=pluginEngine,load_map= True)
  File "D:\Anaconda3\envs\StarCraft\lib\site-packages\sc2reader-1.3.1-py3.6.egg\sc2reader\factories\sc2factory.py", line 85, in load_replay
    return self.load(Replay, source, options, **new_options)
  File "D:\Anaconda3\envs\StarCraft\lib\site-packages\sc2reader-1.3.1-py3.6.egg\sc2reader\factories\sc2factory.py", line 137, in load
    return self._load(cls, resource, filename=filename, options=options)
  File "D:\Anaconda3\envs\StarCraft\lib\site-packages\sc2reader-1.3.1-py3.6.egg\sc2reader\factories\sc2factory.py", line 146, in _load
    obj = cls(resource, filename=filename, factory=self, **options)
  File "D:\Anaconda3\envs\StarCraft\lib\site-packages\sc2reader-1.3.1-py3.6.egg\sc2reader\resources.py", line 288, in __init__
    self.load_map()
  File "D:\Anaconda3\envs\StarCraft\lib\site-packages\sc2reader-1.3.1-py3.6.egg\sc2reader\resources.py", line 405, in load_map
    self.map = self.factory.load_map(self.map_file, **self.opt)
  File "D:\Anaconda3\envs\StarCraft\lib\site-packages\sc2reader-1.3.1-py3.6.egg\sc2reader\factories\sc2factory.py", line 101, in load_map
    return self.load(Map, source, options, **new_options)
  File "D:\Anaconda3\envs\StarCraft\lib\site-packages\sc2reader-1.3.1-py3.6.egg\sc2reader\factories\sc2factory.py", line 136, in load
    resource, filename = self._load_resource(source, options=options)
  File "D:\Anaconda3\envs\StarCraft\lib\site-packages\sc2reader-1.3.1-py3.6.egg\sc2reader\factories\sc2factory.py", line 195, in _load_resource
    contents = self.load_remote_resource_contents(resource, **options)
  File "D:\Anaconda3\envs\StarCraft\lib\site-packages\sc2reader-1.3.1-py3.6.egg\sc2reader\factories\sc2factory.py", line 179, in load_remote_resource_contents
    return urlopen(resource).read()
  File "D:\Anaconda3\envs\StarCraft\lib\urllib\request.py", line 223, in urlopen
    return opener.open(url, data, timeout)
  File "D:\Anaconda3\envs\StarCraft\lib\urllib\request.py", line 532, in open
    response = meth(req, response)
  File "D:\Anaconda3\envs\StarCraft\lib\urllib\request.py", line 642, in http_response
    'http', request, response, code, msg, hdrs)
  File "D:\Anaconda3\envs\StarCraft\lib\urllib\request.py", line 570, in error
    return self._call_chain(*args)
  File "D:\Anaconda3\envs\StarCraft\lib\urllib\request.py", line 504, in _call_chain
    result = func(*args)
  File "D:\Anaconda3\envs\StarCraft\lib\urllib\request.py", line 650, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 400: Bad Request
----------------------------------------------------------------------
Ran 67 tests in 294.889s

FAILED (errors=2, expected failures=3)
```

### PR DESCRIPTION
**Pull request by dneise**
should fix #78

For reviewers:

 - This PR contains some automatic style fixes, using `black -l 127` 
 - Apart from that, the only interesting commits are:
   - a403034cb30fd97b5e7b5b02a106adcf2b29e467
   - 57c7064446c03a2c576bd3cfe73f0dd0595de2b4

### CODE REVIEW

**Comment by dneise** [comment]
Ah so it seems not only did `test_replays/test_all.py` fail. **Also** `test_s2gs/test_all.py` fails. 

So I am also trying to fix that now...

**Comment by dsjoerg** [comment]
I think the s2gs functionality of this library may be safely removed.  We added s2gs parsing back in 2012 because back then s2gs was a valuable source of information about matches for GGTracker.  But the only way to get s2gs files en masse is to automate the SC2 client, and I'm not aware of anyone doing that kind of thing these days.

**Comment by dneise** [comment]
Thanks @dsjoerg for this comment. I propose to have a dedicated issue, in order to discuss the removal of an unused feature. Since the purpose of this PR was to make some existing tests run again.

File: test_replays/test_all.py [hunk 1]
```diff
@@ -7,6 +7,7 @@
 
 # Newer unittest features aren't built in for python 2.6
 import sys
+
 if sys.version_info[:2] < (2, 7):
```
Comment (cclauss) [inline test_replays/test_all.py:11]: Maybe not for this PR but do we really need to support Python versions < 2.7 -- They have been EOL for 5+ years.
https://devguide.python.org/devcycle/#end-of-life-branches

Comment (dneise) [reply test_replays/test_all.py:11]: I personally totally agree here.

Comment (cclauss) [reply test_replays/test_all.py:11]: https://github.com/ggtracker/sc2reader#installation says 2.6 is supported!  My vote would be to drop all legacy Python support.

Comment (dneise) [reply test_replays/test_all.py:11]: Agreed. Should I open a dedicated issue for this? So others can have a word?

Comment (dneise) [reply test_replays/test_all.py:11]: done

**Comment by dneise** [comment]
Should we merge this?

**Comment by cclauss** [comment]
Are the format modification made by [__python/black__](https://github.com/python/black)?

**Comment by dneise** [comment]
yes. Sorry .. was maybe a bit forward of me. Should I remove that?

**Comment by cclauss** [comment]
I am a fan of __python/black__ but others are free to disagree.

**Comment by StoicLoofah** [comment]
Looks good to me. Thanks for fixing this!

I am also okay using black. I'll make an issue for it

**Review by StoicLoofah — APPROVED** [review]

**Pull request merged**